diff --git a/node_modules/react-native-track-player/android/src/main/java/com/doublesymmetry/trackplayer/module/MusicModule.kt b/node_modules/react-native-track-player/android/src/main/java/com/doublesymmetry/trackplayer/module/MusicModule.kt
index b2409a0..a35d05b 100644
--- a/node_modules/react-native-track-player/android/src/main/java/com/doublesymmetry/trackplayer/module/MusicModule.kt
+++ b/node_modules/react-native-track-player/android/src/main/java/com/doublesymmetry/trackplayer/module/MusicModule.kt
@@ -159,7 +159,7 @@ class MusicModule(reactContext: ReactApplicationContext) : ReactContextBaseJavaM
         }
     }
 
-    @ReactMethod
+    @ReactMethod(isBlockingSynchronousMethod = false)
     fun setupPlayer(data: ReadableMap?, promise: Promise) {
         if (isServiceBound) {
             promise.reject(
@@ -250,7 +250,7 @@ class MusicModule(reactContext: ReactApplicationContext) : ReactContextBaseJavaM
         callback.resolve(isServiceBound)
     }
 
-    @ReactMethod
+    @ReactMethod(isBlockingSynchronousMethod = false)
     fun updateOptions(data: ReadableMap?, callback: Promise) = scope.launch {
         if (verifyServiceBoundOrReject(callback)) return@launch
 
@@ -263,7 +263,7 @@ class MusicModule(reactContext: ReactApplicationContext) : ReactContextBaseJavaM
         callback.resolve(null)
     }
 
-    @ReactMethod
+    @ReactMethod(isBlockingSynchronousMethod = false)
     fun add(data: ReadableArray?, insertBeforeIndex: Int, callback: Promise) = scope.launch {
         if (verifyServiceBoundOrReject(callback)) return@launch
 
@@ -307,7 +307,7 @@ class MusicModule(reactContext: ReactApplicationContext) : ReactContextBaseJavaM
         callback.resolve(null)
     }
 
-    @ReactMethod
+    @ReactMethod(isBlockingSynchronousMethod = false)
     fun remove(data: ReadableArray?, callback: Promise) = scope.launch {
         if (verifyServiceBoundOrReject(callback)) return@launch
         val inputIndexes = Arguments.toList(data)
@@ -374,7 +374,7 @@ class MusicModule(reactContext: ReactApplicationContext) : ReactContextBaseJavaM
         callback.resolve(null)
     }
 
-    @ReactMethod
+    @ReactMethod(isBlockingSynchronousMethod = false)
     fun removeUpcomingTracks(callback: Promise) = scope.launch {
         if (verifyServiceBoundOrReject(callback)) return@launch
 
@@ -421,7 +421,7 @@ class MusicModule(reactContext: ReactApplicationContext) : ReactContextBaseJavaM
         callback.resolve(null)
     }
 
-    @ReactMethod
+    @ReactMethod(isBlockingSynchronousMethod = false)
     fun reset(callback: Promise) = scope.launch {
         if (verifyServiceBoundOrReject(callback)) return@launch
 
@@ -432,7 +432,7 @@ class MusicModule(reactContext: ReactApplicationContext) : ReactContextBaseJavaM
         callback.resolve(null)
     }
 
-    @ReactMethod
+    @ReactMethod(isBlockingSynchronousMethod = false)
     fun play(callback: Promise) = scope.launch {
         if (verifyServiceBoundOrReject(callback)) return@launch
 
@@ -440,7 +440,7 @@ class MusicModule(reactContext: ReactApplicationContext) : ReactContextBaseJavaM
         callback.resolve(null)
     }
 
-    @ReactMethod
+    @ReactMethod(isBlockingSynchronousMethod = false)
     fun pause(callback: Promise) = scope.launch {
         if (verifyServiceBoundOrReject(callback)) return@launch
 
@@ -448,7 +448,7 @@ class MusicModule(reactContext: ReactApplicationContext) : ReactContextBaseJavaM
         callback.resolve(null)
     }
 
-    @ReactMethod
+    @ReactMethod(isBlockingSynchronousMethod = false)
     fun stop(callback: Promise) = scope.launch {
         if (verifyServiceBoundOrReject(callback)) return@launch
 
@@ -456,7 +456,7 @@ class MusicModule(reactContext: ReactApplicationContext) : ReactContextBaseJavaM
         callback.resolve(null)
     }
 
-    @ReactMethod
+    @ReactMethod(isBlockingSynchronousMethod = false)
     fun seekTo(seconds: Float, callback: Promise) = scope.launch {
         if (verifyServiceBoundOrReject(callback)) return@launch
 
@@ -480,7 +480,7 @@ class MusicModule(reactContext: ReactApplicationContext) : ReactContextBaseJavaM
         callback.resolve(null)
     }
 
-    @ReactMethod
+    @ReactMethod(isBlockingSynchronousMethod = false)
     fun setVolume(volume: Float, callback: Promise) = scope.launch {
         if (verifyServiceBoundOrReject(callback)) return@launch
 
@@ -495,7 +495,7 @@ class MusicModule(reactContext: ReactApplicationContext) : ReactContextBaseJavaM
         callback.resolve(musicService.getVolume())
     }
 
-    @ReactMethod
+    @ReactMethod(isBlockingSynchronousMethod = false)
     fun setRate(rate: Float, callback: Promise) = scope.launch {
         if (verifyServiceBoundOrReject(callback)) return@launch
 
@@ -545,7 +545,12 @@ class MusicModule(reactContext: ReactApplicationContext) : ReactContextBaseJavaM
         if (verifyServiceBoundOrReject(callback)) return@launch
 
         if (index >= 0 && index < musicService.tracks.size) {
-            callback.resolve(Arguments.fromBundle(musicService.tracks[index].originalItem))
+            val originalItem = musicService.tracks[index].originalItem
+            if (originalItem != null) {
+                callback.resolve(Arguments.fromBundle(originalItem))
+            } else {
+                callback.resolve(null)
+            }
         } else {
             callback.resolve(null)
         }
@@ -584,9 +589,14 @@ class MusicModule(reactContext: ReactApplicationContext) : ReactContextBaseJavaM
         if (verifyServiceBoundOrReject(callback)) return@launch
         callback.resolve(
             if (musicService.tracks.isEmpty()) null
-            else Arguments.fromBundle(
-                musicService.tracks[musicService.getCurrentTrackIndex()].originalItem
-            )
+            else {
+                val originalItem = musicService.tracks[musicService.getCurrentTrackIndex()].originalItem
+                if (originalItem != null) {
+                    Arguments.fromBundle(originalItem)
+                } else {
+                    null
+                }
+            }
         )
     }
 
